core-java-modules/core-java-concurrency-2/src/test/java/com/baeldung/concurrent/completablefuture/CombiningCompletableFuturesUnitTest.java

```diff
@@ -0,0 +1,105 @@
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+
+public class CombiningCompletableFuturesUnitTest {
+
+    @Mock Logger logger;
+
+    @BeforeEach
+    void setup() {
+        logger = mock(Logger.class);
+    }
+
+    private static Stream<Arguments> clientData() {
+        return Stream.of(
+            Arguments.of(List.of("Good Resource"), 1, 0),
+            Arguments.of(List.of("Bad Resource"), 0, 1),
+            Arguments.of(List.of("Good Resource", "Bad Resource"), 1, 1),
+            Arguments.of(List.of("Good Resource", "Bad Resource", "Good Resource", "Bad Resource", "Good Resource"), 3, 2)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("clientData")
+    public void givenMicroserviceClient_whenMultipleCreateResource_thenCombineResults(List<String> inputs, int successCount, int errorCount) {
+        MicroserviceClient mockMicroservice = mock(MicroserviceClient.class);
+        // Return an identifier of 123 on "Good Resource"
+        when(mockMicroservice.createResource("Good Resource"))
+            .thenReturn(CompletableFuture.completedFuture(123L));
+        // Throw an exception on "Bad Resource"
+        when(mockMicroservice.createResource("Bad Resource"))
+            .thenReturn(CompletableFuture.failedFuture(new IllegalArgumentException("Bad Resource")));
+
+        // Given a list of CompletableFutures from our microservice calls...
+        List<CompletableFuture<Long>> clientCalls = new ArrayList<>();
+        for (String resource : inputs) {
+            clientCalls.add(mockMicroservice.createResource(resource));
+        }
+
+        // When all CompletableFutures are completed (exceptionally or otherwise)...
+        Map<Boolean, List<Long>> resultsByValidity = clientCalls.stream()
+            .map(future -> handleFuture(future))
+            .collect(Collectors.partitioningBy(resourceId -> isValidResponse(resourceId)));
+
+        // Then the returned resource identifiers should match what is expected...
+        assertThat(resultsByValidity.getOrDefault(true, List.of()).size()).isEqualTo(successCount);
+        // And the logger mock should be called once for each exception with the expected error message
+        assertThat(resultsByValidity.getOrDefault(false, List.of()).size()).isEqualTo(errorCount);
+        verify(logger, times(errorCount))
+            .error(eq("Encountered error: java.lang.IllegalArgumentException: Bad Resource"));
+    }
+
+    private boolean isValidResponse(long resourceId) {
+        return resourceId != -1L;
+    }
+
+    /**
+     * Completes the given CompletableFuture, handling any exceptions that are thrown.
+     * @param future the CompletableFuture to complete.
+     * @return the resource identifier (-1 if the request failed).
+     */
+    private Long handleFuture(CompletableFuture<Long> future) {
+        return future
+            .exceptionally(ex -> handleError(ex))

```

Comment:
I prefer method references in these kinds of cases, but up to you
```suggestion
            .exceptionally(this::handleError)
```